chain-impl-mockchain/src/vote/manager.rs

```diff
@@ -141,10 +144,13 @@ impl ProposalManager {
         identifier: UnspecifiedAccountIdentifier,
         payload: ValidatedPayload,
     ) -> Result<Self, VoteError> {
-        // we don't mind if we are replacing a vote
-        let votes_by_voters =
-            self.votes_by_voters
-                .insert_or_update_simple(identifier, payload.clone(), |_| Some(payload));
+        // Part of DDoS protection: do not record a new ballot if the account already voted for this
+        // proposal. This protects the system from flooding in a system with cheap/free voting
```

Comment:
Does this on its own provide any advantage for DDoS protection?

Reply:
Not right now, but flooding the system with encrypted votes would quite resource-heavy due to NIZK validation.

Reply:
Another concern there is not getting non-malicious transactions into a block & having the blockchain storage full of garbage.